### ISSUE TITLE
USWDS - Bug: Accessibility Aria Failure on Step Indicator

### ISSUE THREAD

**Issue by jlev123** [uswds/uswds#6113]
### Describe the bug

According to https://www.siteimprove.com/ `aria-label` is a prohibited for use in a `<div>`

All uses of the USWDS Step Indicator are failing accessibility test because of this.

### Steps to reproduce the bug

1. Use the code from USWDS Step Indicator: https://designsystem.digital.gov/components/step-indicator/

![Image](https://github.com/user-attachments/assets/fdcfbad2-b6cf-489f-8f65-610be5d15f3b)

### Expected Behavior

Code should pass accessibility check for all conformance levels

### Related code

<div class="usa-step-indicator" aria-label="progress">

### Screenshots

![Image](https://github.com/user-attachments/assets/25b52df9-8e82-433b-9f55-7a22d63afa6a)

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by amycole501** [comment]
There is a valid progressbar role: https://w3c.github.io/aria/#progressbar,  however that isn't necessarily applicable here as the step indicator isn't exactly operating as a progress bar. 

Currently the step indicator is coded as a list and is announced as a list with X number of items. The labels for each step should be sufficient to allow screen reader users to understand which step they are on, particularly if the role of "current" is used for the current step's page.

**Comment by jlev123** [comment]
`aria-label` can only be used on a `div` if it has a valid `role`.

**Comment by amycole501** [comment]
I can confirm this is showing up as an issue in our version of Siteimprove as well
![Image](https://github.com/user-attachments/assets/39c7d450-63bc-4200-8b5b-cec22f5709f3)

**Comment by amyleadem** [comment]
Thanks for confirming @amycole501. 

I see a couple of low level of effort fixes here, all of which would require a markup change. 

The most straightforward solution I see is to convert the `<div>` here to a landmark `<section>`. I would need to confirm, but this should be a valid element that resolves the SiteImprove issue with minimal impact on the component. 

That said, there is a part of me that wonders about the need for a label that simply reads "progress" here. @amycole501 and/or @alex-hull , can you confirm that we need this `aria-label` here on this component?

**Comment by amycole501** [comment]
Could we call it a "progress list" or "progress steps"? The role of "progressbar" is for the "loading" elements and the step indicator is different and while that's not what we're adding here, it might confuse some SR users as the word "progress" may be familiar to them in another context.

Separately, I do also see value in creating a landmark section as screen reader users can better navigate to it to in SR such as JAWS (you can use shortcut keys to get a list of sections/landmarks, and the user can quickly jump to the step indicator that way). https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/at.html#jaws

**Comment by amyleadem** [comment]
@amycole501 thanks for the response.

I propose that we move forward to fix:
1. Resolve the SiteImprove issue by changing the `<div>` to a `<section>`
2. Create a follow-on issue to re-evaluate the content of the aria-label. If there is not much discussion required, we can try to include it in the same release so that the users can make both markup updates at the same time.

**Comment by amycole501** [comment]
Currently in JAWS I don't hear the ARIA "progress" label announced so it's not doing any good as it is.

If we change the div to a section element, that may then fix the issue in Siteimprove AND allow the ARIA label to be announced.

All that to say yes to #1 and for #2 I think we could leave it off for now. In usability testing with people who are native screen reader users on another project, they didn't have trouble inferring it was a step indicator and knowing how to interact with it. The ARIA label may be superfluous.

**Comment by amyleadem** [comment]
## Update
We decided in an offline meeting to remove the aria-label here. The label is not currently read in JAWS and previous user testing indicated that there is enough semantic structure in the component that there was no need for more context.
